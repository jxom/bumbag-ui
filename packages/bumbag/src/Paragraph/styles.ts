import { cssClass } from '../styled';
import { space, theme } from '../utils';

export const Paragraph = (styleProps) => cssClass`
  &:not(:last-child) {
    margin-bottom: ${space(4)(styleProps)}rem;
  }

  & .bb-Icon {
    vertical-align: -0.125em;
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
