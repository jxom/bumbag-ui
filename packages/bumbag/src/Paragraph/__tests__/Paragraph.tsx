import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Paragraph } from '../Paragraph';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Paragraph>Hello world</Paragraph>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Paragraph color="primary">Hello world</Paragraph>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Paragraph use="div">Hello world</Paragraph>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Paragraph props', () => {
      const { result } = renderHook(() => Paragraph.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Paragraph>{(ParagraphProps) => <div {...ParagraphProps}>Hello world</div>}</Paragraph>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Paragraph.base should render correctly', () => {
    const { container } = render(<Paragraph>hello world</Paragraph>, {
      theme: { Paragraph: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Paragraph>hello world</Paragraph>, {
      theme: { Paragraph: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
