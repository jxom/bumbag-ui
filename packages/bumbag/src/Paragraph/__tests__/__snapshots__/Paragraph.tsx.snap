// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Paragraph"
>
  Hello world
</div>
`;

exports[`composition hook should return with Paragraph props 1`] = `
Object {
  "className": "css-1g8gtm7 css-4lfl7m css-1yuzjtp fp-Box fp-Paragraph",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Paragraph"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

<p
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-Paragraph"
>
  hello world
</p>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Paragraph"
>
  Hello world
</p>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-2 {
  color: #574feb !important;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Paragraph"
>
  Hello world
</p>
`;

exports[`theming Paragraph.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Paragraph"
>
  hello world
</p>
`;
